### PR TITLE
add arn format to the kms policy

### PR DESCRIPTION
**Pull request by rryke**
currently getting:

Error: MalformedPolicyDocumentException: Policy contains a statement with one or more invalid principals.

  on .terraform/modules/flow_logs.kms_key/main.tf line 1, in resource "aws_kms_key" "default":
   1: resource "aws_kms_key" "default" {

## what
* Describe high-level what changed as a result of these commits (i.e. in plain-english, what do these changes mean?)
* Use bullet points to be concise and to the point.

## why
* Provide the justifications for the changes (e.g. business case). 
* Describe why these changes were made (e.g. why do these commits fix the problem?)
* Use bullet points to be concise and to the point.

## references
* Link to any supporting github issues or helpful documentation to add some context (e.g. stackoverflow). 
* Use `closes #123`, if this PR closes a GitHub issue `#123`

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -32,7 +32,7 @@ data "aws_iam_policy_document" "kms" {
       type = "AWS"
 
       identifiers = [
-        "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"
+        "${var.arn_format}::${data.aws_caller_identity.current.account_id}:root"
```
Comment (Gowiem) [inline main.tf:35]: @rryke You might be missing the `:iam` here? We're getting this issue from our tests: 

```
TestExamplesComplete 2021-06-25T02:49:01Z logger.go:66: Error: MalformedPolicyDocumentException: Policy contains a statement with one or more invalid principals.
TestExamplesComplete 2021-06-25T02:49:01Z logger.go:66: 
TestExamplesComplete 2021-06-25T02:49:01Z logger.go:66:   on .terraform/modules/flow_logs.kms_key/main.tf line 1, in resource "aws_kms_key" "default":
TestExamplesComplete 2021-06-25T02:49:01Z logger.go:66:    1: resource "aws_kms_key" "default" {
TestExamplesComplete 2021-06-25T02:49:01Z logger.go:66: 
TestExamplesComplete 2021-06-25T02:49:01Z logger.go:66: 
```

Comment (aknysh) [reply main.tf:35]: you need to add `:iam` to this variable for backwards compatibility and for our tests to work 

here
https://github.com/cloudposse/terraform-aws-vpc-flow-logs-s3-bucket/blob/master/variables.tf#L73

and here
https://github.com/cloudposse/terraform-aws-vpc-flow-logs-s3-bucket/blob/master/examples/complete/variables.tf#L60


and then @rryke  you can specify your own format using the var.

**Review by aknysh — DISMISSED** [review]
thanks @rryke 
please see comments

**Comment by rryke** [comment]
you're right I was a little aggressive with the delete. Added iam back in. Not sure what you meant about specifying my own format. I'm just completing the work that was done previously. As is this doesnt work in govcloud.

**Review by Gowiem — APPROVED** [review]

**Pull request merged**